Tweets.py

```diff
@@ -168,24 +138,21 @@ def run(self):
             return False
 
         # Load new Twitter API keys
-        CONSUMER_KEY = self.dlg.consumer_key.text()
-        CONSUMER_SECRET = self.dlg.consumer_key_secret.text()
-        ACCESS_TOKEN = self.dlg.access_token.text()
-        ACCESS_TOKEN_SECRET = self.dlg.access_token_secret.text()
+        keys = dict()
+        keys['consumer_key'] = self.dlg.consumer_key.text()
+        keys['consumer_secret'] = self.dlg.consumer_key_secret.text()
+        keys['access_token'] = self.dlg.access_token.text()
+        keys['access_token_secret'] = self.dlg.access_token_secret.text()
 
         # If have new Twitter API keys - Save
-        if (CONSUMER_KEY and CONSUMER_SECRET and
-           ACCESS_TOKEN and ACCESS_TOKEN_SECRET):
+        if (keys['consumer_key'] and keys['consumer_secret'] and
+           keys['access_token'] and keys['access_token_secret']):
 
-            with open(self.plugin_dir + '/keys.py', 'w') as keys_file:
-                keys_file.write("# -*- coding: utf-8 -*-\nkeys = dict(\n\
-                consumer_key='" + CONSUMER_KEY + "',\n\
-                consumer_secret='" + CONSUMER_SECRET + "',\n\
-                access_token='" + ACCESS_TOKEN + "',\n\
-                access_token_secret='" + ACCESS_TOKEN_SECRET + "',\n)")
+            with open(os.path.join(self.plugin_dir, 'keys.json'), 'w') as keys_file:
+                json.dump(keys, keys_file, indent=4)
 
         # If haven't new & previous Twitter API keys - Stop
-        elif not all(keys.values()):
+        elif not all(keys.values()) and os.path.isfile('keys.json'):
```

Comment:
I'd wrap `with` section with `try/except` to do the error handling in pythonic style